dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java

```diff
@@ -811,12 +795,12 @@ private void cleanParentChildrenCaches ( Category category ) throws DotDataExcep
                 catCache.removeChildren( parentId );
             }
         }
-        List<String> childrenIds = catCache.getChildren( category );
-        if ( childrenIds != null ) {
-            for ( String childId : childrenIds ) {
-                catCache.removeParents( childId );
-            }
-        }
+		List<Category> children = catCache.getChildren( category );
+		if ( children != null ) {
+			for ( Category child : children ) {
+				catCache.removeParents( child.getCategoryId() );
+			}
+		}
     }
 
 	private class CategoryComparator implements Comparator<Category> {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "CategoryComparator" class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3985)